src/modules/avformat/producer_avformat.c

```diff
@@ -1586,6 +1586,16 @@ static int convert_image_yuvp(producer_avformat self,
                               int dst_full_range)
 {
     int result = self->yuv_colorspace;
+
+    // Workaround: swscale does not convert the range if the src & dst are both 420P
```

Comment:
I think this was recently uncovered because the deinterlace filter is no longer requesting 422 if deinterlacing is not required. I was not able to find a way to get the swscale API to convert between full and mpeg ranges if both source and destination are 420P.